pkg/dockerregistry/server/client/interfaces.go

```diff
@@ -0,0 +1,101 @@
+package client
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kapi "k8s.io/kubernetes/pkg/api"
+
+	imageapiv1 "github.com/openshift/origin/pkg/image/apis/image/v1"
+	userapiv1 "github.com/openshift/origin/pkg/user/apis/user/v1"
+	authapiv1 "k8s.io/kubernetes/pkg/apis/authorization/v1"
+)
+
+type UsersInterfacer interface {
+	Users() UserInterface
+}
+
+type ImagesInterfacer interface {
+	Images() ImageInterface
+}
+
+type ImageSignaturesInterfacer interface {
+	ImageSignatures() ImageSignatureInterface
+}
+
+type ImageStreamImagesNamespacer interface {
+	ImageStreamImages(namespace string) ImageStreamImageInterface
+}
+
+type ImageStreamsNamespacer interface {
+	ImageStreams(namespace string) ImageStreamInterface
+}
+
+type ImageStreamMappingsNamespacer interface {
+	ImageStreamMappings(namespace string) ImageStreamMappingInterface
+}
+
+type ImageStreamSecretsNamespacer interface {
+	ImageStreamSecrets(namespace string) ImageStreamSecretInterface
+}
+
+type ImageStreamTagsNamespacer interface {
+	ImageStreamTags(namespace string) ImageStreamTagInterface
+}
+
+type LimitRangesGetter interface {
+	LimitRanges(namespace string) LimitRangeInterface
+}
+
+type LocalSubjectAccessReviewsNamespacer interface {
+	LocalSubjectAccessReviews(namespace string) LocalSubjectAccessReviewInterface
+}
+
+type SubjectAccessReviews interface {
+	SubjectAccessReviews() SubjectAccessReviewInterface
+}
+
+type ImageSignatureInterface interface {
+	Create(signature *imageapiv1.ImageSignature) (*imageapiv1.ImageSignature, error)
```

Comment:
Need type assertions to make sure that the generated clients conform to these interfaces.

Reply:
+1 good catch